android/app/src/main/java/com/converse/xmtp/Messages.kt

```diff
@@ -238,12 +238,23 @@ suspend fun handleGroupMessage(
     envelope: Envelope,
     remoteMessage: RemoteMessage,
 ): NotificationDataResult {
-    val groups = xmtpClient.conversations.listGroups()
-    val group = groups.find { it.topic == envelope.contentTopic }
+    // No need to sync groups here, because it's already a known group
+    val group = xmtpClient.findGroup(getGroupIdFromTopic(envelope.contentTopic))
     if (group == null) {
         Log.d("PushNotificationsService", "No group found for ${envelope.contentTopic}")
         return NotificationDataResult()
     }
+    sentryTrackMessage(
+        "[NotificationExtension] Found group",
+        mapOf()
+    )
+
+    group.sync()
```

Comment:
I wonder if the error you were seeing earlier was that the message was already processed by the stream. I wonder if  there is a fix here on the sdk side to catch that error and find the message in the database if it's already been processed?

Reply:
Streams are closed when leaving the apo